pywbemtools/pywbemcli/_cmd_connection.py

```diff
@@ -716,8 +716,15 @@ def cmd_connection_save(context, name):
     connections = context.connections_repo
 
     context.spinner_stop()
-
-    connections.add(save_connection.name, save_connection)
+    try:
+        connections.add(save_connection)
+    except Exception as exc:
+        click.echo('Fatal error loading connections file: "{0}". '
```

Comment:
1. The `add()` method can also raise `IOError` when the file cannot be written or renamed to backup, so restricting the message to just loading is misleading in such cases.
2. The broad exception handling is an issue. The `add()` method raises several common exception types such as KeyError, TypeError, IOError (and is documented to raise also ValueError but I did not find that in the code), and that makes it basically impossible to distinguish programming errors from issues with the connections file. I suggest to introduce a proper hirarchy of exceptions for errors related to the connections file, and then a handler like here can simply display the message in the exception without any additional text around it.

Reply:
**Update:** I'll improve the exception handling of `ConnectionRepository` after this PR is merged, so no need to invest in that as part of this PR. You can ignore my comment above.

Reply:
Modified so there is a hiearchy of exceptions and capture at the top level.  Used this to capture all load errors and report them as ConnectionsFileLoadError exception which is subclass of CommenctionsFileError.